### PR TITLE
Add support for a generic preferred citation type

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/views/catalog/_citation.html.erb [hunk 1]
```diff
@@ -8,6 +8,11 @@
   <% @documents&.each do |document| %>
     <h1 class="modal-title"><%= document_heading(document) %></h1>
 
+    <% if document.respond_to?(:export_as_preferred_citation_txt) %>
+      <h2><%= t('blacklight.citation.preferred') %></h2>
+      <%= document.send(:export_as_preferred_citation_txt).html_safe %><br/><br/>
```
Comment (jcoyne) [inline app/views/catalog/_citation.html.erb:13]: why is the `send` needed?

Comment (cbeer) [reply app/views/catalog/_citation.html.erb:13]: No idea, but it's the pattern we've established. I'm happy to kill it, but maybe it's a 8.0 thing?

Comment (jcoyne) [reply app/views/catalog/_citation.html.erb:13]: I think `document.export_as_preferred_citation_txt.html_safe` would be idea.

**Comment by barmintor** [comment]
It seems like at this point - with the logic from `app/views/catalog/_citation.html.erb` moved into `citation_component.html.erb` and now with a somewhat different implementation, this PR would need to be re-implemented, right?

**Comment by barmintor** [comment]
Since this can no longer be merged, I have ticketed it for further discussion of the feature in #2704

**Pull request closed without merging**